server/StudySharp.API/Controllers/CourseController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using StudySharp.API.Requests.Courses;
+using StudySharp.API.Responses.Courses;
+using StudySharp.ApplicationServices.Commands;
+using StudySharp.ApplicationServices.Queries;
+using StudySharp.Domain.General;
+
+namespace StudySharp.API.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("api/courses")]
+
+    public class CourseController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        private readonly IMapper _mapper;
+
+        public CourseController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpPost]
+        public async Task<OperationResult> Add([FromBody] AddCourseRequest addCourseRequest)
+        {
+            var addCourseCommand = _mapper.Map<AddCourseCommand>(addCourseRequest);
+            return await _mediator.Send(addCourseCommand);
+        }
+
+        [HttpDelete("{id:int}")]
+        public async Task<OperationResult> Remove([FromRoute] RemoveCourseByIdRequest removeCourseByIdRequest)
+        {
+            var removeCourseByIdCommand = _mapper.Map<RemoveCourseByIdCommand>(removeCourseByIdRequest);
+            return await _mediator.Send(removeCourseByIdCommand);
+        }
+
+        [HttpGet("{id:int}")]
+        public async Task<OperationResult<GetCourseByIdResponse>> GetCourseById([FromRoute] GetCourseByIdRequest getCourseByIdRequest)
```

Comment:
"Microsoft.AspNetCore.Routing.Matching.AmbiguousMatchException: The request matched multiple endpoints. Matches: \r\n\r\nStudySharp.API.Controllers.CourseController.GetCourseById (StudySharp.API)\r\nStudySharp.API.Controllers.CourseController.GetCoursesByTeacherId (StudySharp.API)